### PR TITLE
add transcript for episode 031 (Rust 1.46+1.47)

### PR DESCRIPTION
**Pull request by ericseppanen**

### CODE REVIEW

File: _transcripts/031-rust-1.46-1.47.md [hunk 1]
```diff
@@ -0,0 +1,894 @@
...
+that if you have an array, that is of some length `N`, you might want to have
+implementations that are generic over that number `,`. And in theory, you could
+have other types that are also generic over numbers, and not just over types.
+Now, the primary way in which const generics is currently exposed to the world
+is through implementations on arrays. So the idea is that if you have an array
+of some length, you might want it to implement, say, `Debug`, no matter how long
+the array is. And it used to be that the Rust compiler would pull this trick of,
+it just copy-pasted the implementation, like, 32 times and just implemented,
+like `impl Debug for [T; 0]`. And then it did the same for `[T; 1]`, `[T; 2]`.
+And it just repeated this down.
```
Comment (ericseppanen) [inline _transcripts/031-rust-1.46-1.47.md:398]: Please review my translation of speech to Rust?

File: _transcripts/031-rust-1.46-1.47.md [hunk 2]
```diff
@@ -0,0 +1,894 @@
...
+`Default` trait.
+
+__Ben__: Which is totally reasonable, in a vacuum.
+
+__Jon__: Yeah, because you don't have any `T`s, so it doesn't matter if `T`
+implements `Default`. Unfortunately, there isn't, at least far as I'm aware, a
+way to express this in today's const generics, right? If you imagine that you
+write like `impl<T, const N> Default for [T; N]`. You don't have a way to say,
+like, require `Default` for `T` if `N` is greater than one, or greater than or
+equal to one, but don't require it if `N` is 0. You can't currently write that.
```
Comment (ericseppanen) [inline _transcripts/031-rust-1.46-1.47.md:462]: Please review my translation of speech to Rust? (audio at 26:38)

What was actually spoken was "impl T and const generic N..." so I'm guessing this is what was meant.

File: _transcripts/031-rust-1.46-1.47.md [hunk 3]
```diff
@@ -0,0 +1,894 @@
...
+There were a bunch of smaller library changes in 1.47 as well. And I think
+before we go to some of the specifics, do you want to talk a little bit about
+the `const` changes here?
+
+__Ben__: Yeah. So as we mentioned before in the previous release, `const` now
+has the capability of doing a bunch of `if`s and `loop`s. And so there are
+plenty of newly-const methods. In this case, although the scope is pretty
+narrow, the biggest group are probably: for all integers, there are various
+operations. There's `checked_add`, `checked_sub`-traction,
+`checked_mul`-tiplication, and by "checked" we mean that in this case, if it
```
Comment (ericseppanen) [inline _transcripts/031-rust-1.46-1.47.md:615]: My translation of "checked subtraction, checked multiplication" may be questionable here? (audio 34:56)

Maybe I should have just dropped the -traction, -tiplication instead?

File: _transcripts/031-rust-1.46-1.47.md [hunk 4]
```diff
@@ -0,0 +1,894 @@
...
+`saturating_sub`, etc. In this case, if it would overflow, it will— or under
+flow as well, I'm pretty sure, I guess in subtraction. I guess you can also add
+a negative in the signed case.
+
+__Jon__: Yeah, it's—
+
+__Ben__: But yeah, in either case, instead of overflowing, they will just kind
+of clamp themselves to the highest value. And so, if you try to add two `MAX`
+ints, you just get `MAX` int back. Or try to subtract zero or, you know, like a
+`MAX` int from zero, you just get zero back. And so different people want
```
Comment (ericseppanen) [inline _transcripts/031-rust-1.46-1.47.md:626]: `MAXINT`? `MAX_INT`? I'm not sure how to render this, so I guessed at something vaguely Rust-y (audio 35:22).

File: _transcripts/031-rust-1.46-1.47.md (hunk 3 above)
Comment (jonhoo) [reply _transcripts/031-rust-1.46-1.47.md:615]: No, I like it!

File: _transcripts/031-rust-1.46-1.47.md (hunk 4 above)
Comment (jonhoo) [reply _transcripts/031-rust-1.46-1.47.md:626]: I think the way you did it is probably just as understandable as the alternatives, and looks slightly nicer :sweat_smile:

**Review by jonhoo — APPROVED** [review]

**Pull request merged**